src/theory/splitter.cpp

```diff
@@ -0,0 +1,181 @@
+/*
+ * Top contributors (to current version):
+ *   Amalee Wilson, Andrew Wu
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * Splitter for creating partitions.
+ */
+
+#include "theory/splitter.h"
+
+#include <math.h>
+
+#include "expr/node_algorithm.h"
+#include "expr/node_builder.h"
+#include "options/smt_options.h"
+#include "prop/prop_engine.h"
+#include "prop/theory_proxy.h"
+#include "prop/zero_level_learner.h"
+#include "theory/theory_engine.h"
+#include "theory/theory_id.h"
+#include "theory/theory_traits.h"
+
+using namespace std;
+using namespace cvc5::theory;
+
+namespace cvc5 {
+
+namespace theory {
+Splitter::Splitter(TheoryEngine* theoryEngine, prop::PropEngine* propEngine)
+    : d_numPartitions(options::computePartitions()),
+      d_numChecks(0),
+      d_numPartitionsSoFar(0),
+      d_partitionFile(options::writePartitionsToFileName())
+{
+  d_valuation = std::make_unique<Valuation>(theoryEngine);
+  d_propEngine = propEngine;
+  d_output = &std::cout;
+  if (d_partitionFile != "")
+  {
+    d_partitionFileStream.open(d_partitionFile);
+    d_output = &d_partitionFileStream;
+    d_partitionFileStream.close();
```

Comment:
Isn't `d_output` always not usable after that?

Reply:
Yeah, actually I managed to break writing it to file altogether, oops.